static/js/containers/PaymentPage.js

```diff
@@ -0,0 +1,162 @@
+// @flow
+/* global SETTINGS: false */
+import React from 'react';
+import { connect } from 'react-redux';
+import type { Dispatch } from 'redux';
+import _ from 'lodash';
+import R from 'ramda';
+
+import {
+  clearUI,
+  setPaymentAmount,
+  setSelectedKlassIndex
+} from '../actions';
+import { actions } from '../rest';
+import { createForm } from '../util/util';
+import Payment from '../components/Payment';
+import PaymentHistory from '../components/PaymentHistory';
+import type { UIState } from '../reducers';
+import type { RestState } from '../rest';
+import type { InputEvent } from '../flow/events';
+
+class PaymentPage extends React.Component {
+  props: {
+    dispatch: Dispatch,
+    ui: UIState,
+    payment: RestState,
+    klasses: RestState,
+    selectedKlass: Object,
+  };
+
+  fetchAPIdata() {
+    this.fetchKlasses();
+  }
+
+  componentDidMount() {
+    this.fetchAPIdata();
+  }
+
+  componentDidUpdate() {
+    this.fetchAPIdata();
+  }
+
+  componentWillUnmount() {
+    const { dispatch } = this.props;
+    dispatch(clearUI());
+  }
+
+  fetchKlasses() {
+    const { dispatch, klasses } = this.props;
+    if (!klasses.fetchStatus) {
+      dispatch(
+        actions.klasses({username: SETTINGS.user.username})
+      );
+    }
+  }
+
+  sendPayment = () => {
+    const {
+      dispatch,
+      ui: { paymentAmount },
+      selectedKlass
+    } = this.props;
+    if (selectedKlass && paymentAmount) {
+      dispatch(
+        actions.payment({
+          paymentAmount: paymentAmount,
+          klassKey: selectedKlass.klass_key
+        })
+      ).then(result => {
+        const {url, payload} = result;
+        const form = createForm(url, payload);
+        const body = document.querySelector("body");
+        body.appendChild(form);
+        form.submit();
+      });
+    }
+  };
+
+  setPaymentAmount = (event: InputEvent) => {
+    const { dispatch } = this.props;
+    dispatch(setPaymentAmount(event.target.value));
+  };
+
+  setSelectedKlassIndex = (event: InputEvent) => {
+    const { dispatch } = this.props;
+    dispatch(setSelectedKlassIndex(parseInt(event.target.value)));
+  };
+
+  hasPastPayments = () => {
+    const { klasses } = this.props;
+
+    if (!klasses.data) return false;
+
+    let totalPaid = R.compose(
+      R.sum,
+      R.map(
+        R.propOr(0, 'total_paid')
+      )
+    )(klasses.data);
+    return totalPaid > 0;
+  };
+
+  render() {
+    const {
+      ui,
+      payment,
+      klasses,
+      selectedKlass
+    } = this.props;
+
+    let renderedPaymentHistory = this.hasPastPayments() ?
+      <div className="body-row">
+        <PaymentHistory klasses={klasses} />
+      </div> :
+      null;
```

Comment:
makes sense. i wasn't sure about this at first since the true/false cases can be multi-line (as it is here) but parenthesis can help with that. let me know what you think